tests/test_flake8_trio.py

```diff
@@ -65,19 +69,40 @@ def check_version(test: str):
 }
 
 
+def check_autofix(test: str, plugin: Plugin, generate_autofix: bool):
+    if test not in autofix_files:
+        return
+    checked_autofix.add(test)
+    visited_code = plugin.module.code
+    current_generated = autofix_files[test].read_text()
+    if generate_autofix:
+        if visited_code != current_generated:
+            print(f"regenerating {test}")
+        autofix_files[test].write_text(visited_code)
+        return
+    assert visited_code == current_generated
+
+
 @pytest.mark.parametrize(("test", "path"), test_files)
-def test_eval(test: str, path: Path):
+def test_eval(test: str, path: Path, generate_autofix: bool):
     content = path.read_text()
     if "# NOTRIO" in content:
         pytest.skip("file marked with NOTRIO")
 
     expected, parsed_args = _parse_eval_file(test, content)
+    parsed_args.append("--autofix")
     if "# TRIO_NO_ERROR" in content:
         expected = []
 
     plugin = Plugin.from_source(content)
     _ = assert_expected_errors(plugin, *expected, args=parsed_args)
 
+    check_autofix(test, plugin, generate_autofix)
+
+
+def test_assure_all_autofix_files_used():
+    assert set(autofix_files.keys()) - checked_autofix == set()
```

Comment:
this one is a minor pain since I usually develop locally with `--failed-first`, but didn't see any simple way of forcing this to be run late without installing pytest plugins (and that might still get overridden by `--failed-first`). But should help the occasional messed up file management by making sure all files in autofixed_files are used.
(although ig that could be done by static analysis of directory contents....)

Reply:
fixed, now statically checking the contents of the file lists.